### PR TITLE
feat: implement wasm feature from config

### PR DESCRIPTION
**Pull request by bkioshn**
## Describe your changes
Implement WASM feature from config

Remove MAX_FILE_SIZE and use storeCodeMaxFileSize
When Wasm flag is disabled, the following path should not be accessed
- /codes
- /contract-list
- /deploy 
- /execute
- /instantiate
- /migrate
- /public-project
- /query
- /recent-codes
- /upload
- /contract 
- /contract-list/instantiated-by-me
- /contract-list/saved-contracts
- /contract-list/[name]
- /admin

### CODE REVIEW

**Review by poomthiti — APPROVED** [review]

**Comment by poomthiti** [comment]
I think we should create a custom hook for checking wasm path redirect and amptrack. Maybe add a refactor task later @evilpeach

File: src/lib/pages/codes/index.tsx [hunk 1]
```diff
@@ -59,8 +60,9 @@ const Codes = observer(() => {
   const isSearching = !!keyword || permissionValue !== "all";
 
   useEffect(() => {
-    if (router.isReady) AmpTrack(AmpEvent.TO_MY_CODES);
-  }, [router.isReady]);
+    if (!wasm.enabled) navigate({ pathname: "/", replace: true });
+    else if (router.isReady) AmpTrack(AmpEvent.TO_MY_CODES);
```
Comment (songwongtp) [inline src/lib/pages/codes/index.tsx:64]: ditto

**Review by songwongtp — CHANGES_REQUESTED** [review]

File: src/lib/pages/deploy/index.tsx [hunk 2]
```diff
@@ -26,10 +27,11 @@ const Deploy = () => {
   const router = useRouter();
   const navigate = useInternalNavigate();
   const selectChain = useSelectChain();
+  useWasmConfig({ shouldRedirect: true });
 
   useEffect(() => {
     if (router.isReady) AmpTrack(AmpEvent.TO_DEPLOY);
-  }, [router.isReady]);
+  }, [navigate, router.isReady]);
```
Comment (evilpeach) [inline src/lib/pages/deploy/index.tsx:34]: ditto

File: src/lib/pages/contract-list/slug.tsx [hunk 3]
```diff
@@ -30,7 +30,9 @@ import { AmpEvent, AmpTrack } from "lib/services/amplitude";
 import type { ContractAddr } from "lib/types";
 import { formatSlugName, getFirstQueryParam } from "lib/utils";
 
+// eslint-disable-next-line sonarjs/cognitive-complexity
```
Comment (evilpeach) [inline src/lib/pages/contract-list/slug.tsx:33]: can we remove?

File: src/lib/pages/code-details/index.tsx [hunk 4]
```diff
@@ -123,13 +124,15 @@ const CodeDetailsBody = observer(
...
+  useWasmConfig({ shouldRedirect: true });
   const router = useRouter();
+  const navigate = useInternalNavigate();
   const codeIdParam = getFirstQueryParam(router.query.codeId);
   const data = useCodeData(codeIdParam);
 
   useEffect(() => {
     if (router.isReady) AmpTrack(AmpEvent.TO_CODE_DETAIL);
-  }, [router.isReady]);
+  }, [navigate, router.isReady]);
```
Comment (evilpeach) [inline src/lib/pages/code-details/index.tsx:135]: remove

File: src/lib/pages/contract-details/index.tsx [hunk 5]
```diff
@@ -144,7 +150,7 @@ const ContractDetails = observer(() => {
 
   useEffect(() => {
     if (router.isReady) AmpTrack(AmpEvent.TO_CONTRACT_DETAIL);
-  }, [router.isReady]);
+  }, [navigate, router.isReady]);
```
Comment (evilpeach) [inline src/lib/pages/contract-details/index.tsx:153]: remove

**Review by evilpeach — CHANGES_REQUESTED** [review]

**Review by evilpeach — APPROVED** [review]

**Pull request merged**